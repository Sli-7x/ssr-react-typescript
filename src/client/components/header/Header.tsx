import * as React from 'react';
import { Link } from 'react-router-dom';

export interface IHeaderProps {}

export class Header extends React.Component<IHeaderProps, any> {
  render() {
    return (
      <div id="header">
        <Link to="/"> Home </Link>
        <Link to="/about"> About </Link>
        <Link to="/product"> Product </Link>
      </div>
    );
  }
}
