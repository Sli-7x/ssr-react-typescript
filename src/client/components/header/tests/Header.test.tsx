import * as React from 'react';
import { shallow } from 'enzyme';
import Header from '../Header';
import * as renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router';

describe('Footer.tsx', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<StaticRouter context={{}}><Header /></StaticRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render good', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find('#header').length).toBe(1);
  });
});
