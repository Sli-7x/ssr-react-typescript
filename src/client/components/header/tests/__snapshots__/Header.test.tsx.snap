// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer.tsx should render correctly 1`] = `
<div
  id="header"
>
  <a
    href="/"
    onClick={[Function]}
  >
     Home 
  </a>
  <a
    href="/about"
    onClick={[Function]}
  >
     About 
  </a>
  <a
    href="/product"
    onClick={[Function]}
  >
     Product 
  </a>
</div>
`;
